core/src/test/scala/unit/kafka/server/GroupCoordinatorBaseRequestTest.scala

```diff
@@ -934,20 +937,10 @@ class GroupCoordinatorBaseRequestTest(cluster: ClusterInstance) {
     )
   }
 
-  protected def connectAndReceiveWithoutClosingSocket[T <: AbstractResponse](
+  protected def sendAndReceiveFromExistingSocket[T <: AbstractResponse](
     request: AbstractRequest,
-    destination: Int
-  )(implicit classTag: ClassTag[T]): T = {
-    val socket = IntegrationTestUtils.connect(brokerSocketServer(destination), cluster.clientListener())
-    openSockets += socket
-    IntegrationTestUtils.sendAndReceive[T](request, socket)
-  }
-
-  protected def connectAndReceiveWithoutClosingSocket[T <: AbstractResponse](
-    request: AbstractRequest
+    socket: Socket
   )(implicit classTag: ClassTag[T]): T = {
-    val socket = IntegrationTestUtils.connect(cluster.anyBrokerSocketServer(), cluster.clientListener())
-    openSockets += socket
     IntegrationTestUtils.sendAndReceive[T](request, socket)
   }
```

Comment:
Why do you need this method and can't use `IntegrationTestUtils.sendAndReceive[T](request, socket)` directly?